### PR TITLE
docs: Update database startup machine diagram

### PR DESCRIPTION
**Pull request by alamb**
# Rationale
As [noted](https://github.com/influxdata/influxdb_iox/pull/3292#pullrequestreview-826159666
) by @crepererum , the diagram (well textual description really) has drifted from the reality

# Changes
Update the docstring of `DatabaseState` enum

### CODE REVIEW

File: server/src/database.rs [hunk 1]
```diff
@@ -1061,23 +1061,45 @@ pub enum InitError {
 /// The Database startup state machine
 ///
 /// A Database starts in DatabaseState::Known and advances through the
-/// states in sequential order until it reaches Initialized or an error
-/// is encountered.
+/// non error states in sequential order until either:
```
Comment (alamb) [inline server/src/database.rs:1064]: While a diagram might be nice here I worry about making one and then having it drift from reality as the state machine is changed.

Thus I tried to add some helpful, though somewhat vague description here.

File: server/src/database.rs [hunk 2]
```diff
@@ -1061,23 +1061,45 @@ pub enum InitError {
...
+///
+/// 1. It reaches `Initialized`
+///
+/// 2. It is reset to `Known` and starts initialization again
+///
+/// 3. An error is encountered, in which case it transitions to one of
+/// the error states. Most are Terminal (and thus require operator
+/// intervention) but some (such as `WriteBufferCreationError` may
```
Comment (carols10cents) [inline server/src/database.rs:1072]: Unclosed parentheses on line 1072:

```suggestion
/// intervention) but some (such as `WriteBufferCreationError`) may
```

**Review by carols10cents — APPROVED** [review]
LGTM, one nit!

File: server/src/database.rs [hunk 3]
```diff
@@ -1061,23 +1061,45 @@ pub enum InitError {
...
-/// is encountered.
+/// non error states in sequential order until either:
+///
+/// 1. It reaches `Initialized`
+///
+/// 2. It is reset to `Known` and starts initialization again
+///
+/// 3. An error is encountered, in which case it transitions to one of
+/// the error states. Most are Terminal (and thus require operator
+/// intervention) but some (such as `WriteBufferCreationError`) may
```
Comment (crepererum) [inline server/src/database.rs:1072]: This will change once I address #3335 but for now it's correct.

**Review by crepererum — APPROVED** [review]

**Pull request merged**